benchmarktests/target_auth_gcp.go

```diff
@@ -0,0 +1,294 @@
+	}
+	return nil
+}
+
+func (g *GCPAuth) GetTargetInfo() TargetInfo {
+	return TargetInfo{
+		method:     GCPAuthTestMethod,
+		pathPrefix: g.pathPrefix,
+	}
+}
+
+func (g *GCPAuth) Setup(client *api.Client, randomMountName bool, mountName string) (BenchmarkBuilder, error) {
+	var err error
+	authPath := mountName
+	config := g.config.Config
+	g.logger = targetLogger.Named(GCPAuthTestType)
+
+	if randomMountName {
+		authPath, err = uuid.GenerateUUID()
+		if err != nil {
+			return nil, fmt.Errorf("can't generate UUID for mount name: %v", err)
+		}
+	}
+
+	g.logger.Trace(mountLogMessage("auth", "gcp", authPath))
+	err = client.Sys().EnableAuthWithOptions(authPath, &api.EnableAuthOptions{
+		Type: "gcp",
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error enabling gcp: %v", err)
+	}
+	setupLogger := g.logger.Named(authPath)
+
+	// check if the provided argument should be read from file
+	creds := config.GCPAuthConfig.Credentials
+	if len(creds) > 0 && creds[0] == '@' {
+		var err error
+		contents, err := ioutil.ReadFile(creds[1:])
+		if err != nil {
+			return nil, fmt.Errorf("error reading file: %w", err)
+		}
+
+		config.GCPAuthConfig.Credentials = string(contents)
+	}
+
+	setupLogger.Trace(parsingConfigLogMessage("gcp auth"))
+	GCPAuthConfig, err := structToMap(config.GCPAuthConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing gcp auth config from struct: %v", err)
+	}
+
+	// Write GCP config
+	setupLogger.Trace(writingLogMessage("gcp auth config"))
+	_, err = client.Logical().Write("auth/"+authPath+"/config", GCPAuthConfig)
+
+	if err != nil {
+		return nil, fmt.Errorf("error writing gcp config: %v", err)
+	}
+
+	setupLogger.Trace(parsingConfigLogMessage("role"))
+	GCPRoleConfig, err := structToMap(config.GCPTestRoleConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing role config from struct: %v", err)
+	}
+
+	// Write GCP Role
+	setupLogger.Trace(writingLogMessage("role"), "name", config.GCPTestRoleConfig.Name)
+	_, err = client.Logical().Write("auth/"+authPath+"/role/"+config.GCPTestRoleConfig.Name, GCPRoleConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error writing gcp role: %v", err)
+	}
+
+	// Select one of the configured service accounts if more than 1
+	rand.Seed(time.Now().Unix())
+
+	var serviceAccount string
+
+	if len(config.GCPTestRoleConfig.BoundServiceAccounts) > 0 {
+		n := rand.Int() % len(config.GCPTestRoleConfig.BoundServiceAccounts)
+		serviceAccount = config.GCPTestRoleConfig.BoundServiceAccounts[n]
+	}
+
+	iam := config.GCPTestRoleConfig.Type == "iam"
+	jwt, err := getSignedJwt(g.config.Config.GCPTestRoleConfig.Name, config.GCPAuthConfig.Credentials, config.GCPTestRoleConfig.MaxJWTExp, serviceAccount, iam)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching JWT: %v", err)
+	}
+
+	return &GCPAuth{
+		header:     generateHeader(client),
+		pathPrefix: "/v1/" + filepath.Join("auth", authPath),
+		jwt:        jwt,
+		roleName:   g.config.Config.GCPTestRoleConfig.Name,
```

Comment:
Nit: You set a `config` var at the top (https://github.com/hashicorp/vault-benchmark/pull/131/files#diff-cd2dfbf07314c28343239c0a14e1ca3bde2e051fc0be649ad1fb896a6146cf8eR141), maybe just use that here since its a bit neater?